From 76a32d78bd6e459c38ad7774b92e8ff8dec48881 Mon Sep 17 00:00:00 2001
From: stickz <stickman002@mail.com>
Date: Fri, 24 Feb 2023 13:29:38 -0500
Subject: [PATCH] Allow 10 gigabit speed throttles

This commit increases max upload and download speed by 16 times. It allows utilization of 10 gigabit connections.
---
 src/torrent/download/resource_manager.cc | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/torrent/download/resource_manager.cc b/src/torrent/download/resource_manager.cc
index bc6374d26..153ca854f 100644
--- a/src/torrent/download/resource_manager.cc
+++ b/src/torrent/download/resource_manager.cc
@@ -266,16 +266,16 @@ ResourceManager::set_group(iterator itr, uint16_t grp) {
 
 void
 ResourceManager::set_max_upload_unchoked(unsigned int m) {
-  if (m > (1 << 16))
-    throw input_error("Max unchoked must be between 0 and 2^16.");
+  if (m > (1 << 20))
+    throw input_error("Max unchoked must be between 0 and 2^20.");
 
   m_maxUploadUnchoked = m;
 }
 
 void
 ResourceManager::set_max_download_unchoked(unsigned int m) {
-  if (m > (1 << 16))
-    throw input_error("Max unchoked must be between 0 and 2^16.");
+  if (m > (1 << 20))
+    throw input_error("Max unchoked must be between 0 and 2^20.");
 
   m_maxDownloadUnchoked = m;
 }
